scripts/bootstrap_dsh.py

```
@@ -0,0 +1,398 @@
+
+
+def validate_project_paths(project_root: Path) -> None:
+    checks = [
+        project_root / SKILL_RELATIVE,
+        project_root / OVERLAY_RELATIVE,
+        (project_root / OVERLAY_RELATIVE).parent,
+    ]
+    for path in checks:
+        resolved = path.resolve()
+        if not path_is_relative_to(resolved, project_root):
+            raise RuntimeError(f"Refusing to write outside project root via resolved path: {path} -> {resolved}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)